.github/workflows/flowzone.yml

```diff
@@ -1351,10 +1355,7 @@ jobs:
       - custom_publish
       - custom_finalize
       - custom_clean
-    if: |
-      !failure() &&
-      needs.event_types.result == 'success' &&
-      needs.project_types.outputs.custom_always == 'true'
```

Comment:
Was this change committed by mistake?

Reply:
@klutchell this is intentional, the current check still skips the always step when the pipeline is cancelled/fails.

I think it should be a very simple check instead:

```
if: needs.project_types.outputs.custom_always == 'true'
```

.. with the caveat that the always custom action should return `true` even if the cleanup fails.

Reply:
basically, if the user has `always/action.yml`, we *always* need to run it, since it's important (otherwise expensive assets maybe left running).

Reply:
What if a previous step fails? Do we still want/need to run the custom step?

At the very least this change should be a separate commit, if not a separate PR. It's very confusing otherwise.

Reply:
> What if a previous step fails? Do we still want/need to run the custom step?

that's the whole point of `always()` - if something fails, we still want to ensure the clen-up steps are done. This would be ideally handled in the composite action, but the GH approach is to kill them without respecting their own `always()` steps.

Reply:
> At the very least this change should be a separate commit, if not a separate PR. It's very confusing otherwise.

@klutchell agree, this is a shortcut on my part, I'll move it